File: Week_01/rotate_array.go
func rotate(nums []int, k int)  {
    k = k % len(nums)
    reverse(nums)
    reverse(nums[:k])
    reverse(nums[k:])
}

func reverse(nums []int) {
    for i := 0; i < len(nums)/2; i++ {
        nums[i], nums[len(nums)-i-1] = nums[len(nums)-i-1], nums[i]
    }
}

File: Week_01/rotate_array.py
# timeout
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        for i in xrange(k):
            temp = nums[-1]
            for m in range(len(nums), 0, -1):
                m = m - 1
                if m == 0:
                    nums[m] = temp
                else:
                    nums[m] = nums[m-1]


